engine/core/src/main/java/org/datacleaner/components/composition/AbstractWrappedAnalysisJobTransformer.java

```diff
@@ -133,7 +133,7 @@ public OutputColumns getOutputColumns() {
         final String[] names = new String[size];
         final Class<?>[] types = new Class[size];
         for (int i = 0; i < size; i++) {
-            InputColumn<?> outputColumn = _outputColumns.get(0);
+            final InputColumn<?> outputColumn = _outputColumns.get(i);
```

Comment:
For reference - this was the "sincere" bug - it wasn't applying the right index.